### DISCUSSION TITLE
SMS登录问题修复方案

### DISCUSSION THREAD

**Discussion by michael-eddy** [ywmoyue/biliuwp-lite#116] [feature]
伪造buvid
`static buvid(): string {`
` var mac = [];`
`for (let i = 0; i < 6; i++) {`
`var min = Math.min(0, 0xff)`
` var max = Math.max(0, 0xff)`
` var num = parseInt((Math.random() * (min - max + 1) + max).toString()).toString(16)`
`mac.push(num)`
`}`
` var md5 = this.md5(mac.join(':'));`
`var md5Arr = md5.split('');`
`return "XY${md5Arr[2]}${md5Arr[12]}${md5Arr[22]}${md5}"`
`}`

调用接口
`&channel=bili&cid=&tel=&login_session_id=&statistics=' + encodeURIComponent('{"appId":1,"platform":3,"version":"7.27.0","abtest":""}')
+ '&buvid=&local_id=' + buvid + '&gee_seccode=&gee_validate=&gee_challenge=&recaptcha_token=`

主要修改一下数据
`channel=bili`
`&statistics={"appId":1,"platform":3,"version":"7.27.0","abtest":""}   key-value时使用字符串格式否则需要使用URL编码`
`&buvid=`
`&local_id=  也是上面的 buvid`

**Comment by ywmoyue** [comment]
感谢您提供的方案，但我没太看明白，是除了加buvid还要改什么别的参数吗
我试了下加上buvid就能成功登录了

**Reply by michael-eddy** [reply]
其他貌似加不加都影響不大，但是不排除改天又需要了
